cmd/snap/cmd_run.go

```diff
@@ -579,23 +590,47 @@ func (x *cmdRun) snapRunApp(snapApp string, args []string) error {
 	}
 }
 
-func (x *cmdRun) snapRunHook(snapName string) error {
+func (x *cmdRun) snapRunHook(snapTarget string) error {
+	snapInstance, componentName := snap.SplitSnapComponentInstanceName(snapTarget)
+
 	revision, err := snap.ParseRevision(x.Revision)
 	if err != nil {
 		return err
 	}
 
-	info, err := getSnapInfo(snapName, revision)
+	info, err := getSnapInfo(snapInstance, revision)
 	if err != nil {
 		return err
 	}
 
-	hook := info.Hooks[x.HookName]
+	var (
+		hook      *snap.HookInfo
+		component *snap.ComponentInfo
+	)
+	if componentName == "" {
+		hook = info.Hooks[x.HookName]
+	} else {
+		// TODO: we need to figure out how to get the component revision to set
+		// the environment variables that we provide to the hook
+		componentRevision := snap.Revision{}
+		_ = componentRevision
```

Comment:
Maybe we need a `--comprev` argument as we are using `-r` for the snap revision for hooks, and use it in `runHookAndWait`. Not sure though why we do that instead of reading the snap information as in the non-hook case, probably @pedronis knows.

Reply:
there was the idea to run hooks for the non-current revisions but I'm not sure it can work in practice. maybe we should revisit this

Reply:
Sorry, this TODO is old and I forgot to remove it. We are able to get the revision of the component ever since the change that @alfonsosanchezbeato made with the component symlinks. Although yes, I'm not sure what to do with running component hooks for revisions of a component that is not the current one. I'm not sure how that would work? What is the use case for that?